src/weights/weights.cpp

```diff
@@ -254,6 +249,25 @@ void weights::set_matrix_distribution(El::DistData dist) {
   m_matrix_dist = dist;
 }
 
+void weights::set_comm(lbann_comm& comm) {
```

Comment:
Two things: 
1. This should probably be a `shared_ptr`.
2. Should this update the matrix distribution?